src/deepsparse/utils/timer.py

```diff
@@ -114,6 +114,26 @@ def has_stage(self, stage: str) -> bool:
         """
         return stage in self.stages
 
+    @contextmanager
+    def time(self, stage: str):
+        """
+        Context Manager to record the time for a stage in the given context
+
+        example:
+        ```
+        with timer.time(STAGE_NAME):
+            # do something...
+        ```
+
+        :param stage: the name of the stage to time
+        """
+        self.start(stage)
+
+        try:
+            yield
+        finally:
+            self.stop(stage)
```

Comment:
The implementation looks good but I don't believe we need finally here;
```suggestion
        self.start(stage)
        yield
        self.stop(stage)
```

Using finally with yield can sometimes cause unforeseen errors, (like executing the finally clause if the generator reference is not stored) more info here: https://stackoverflow.com/a/56063376/21911160

Though using the `@contextmanager` decorator protects us from executing the finally clause(as it stores a reference to the generator) but I think the proposed version is safer and reads better (also reduces an indentation level)